src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -1999,6 +1999,72 @@ public static void GetSystemTimeZones()
             }
         }
 
+        private static void ValidateTimeZonesSorting(ReadOnlyCollection<TimeZoneInfo> zones)
+        {
+            // validate sorting: first by base offset, then by display name
+            for (int i = 1; i < zones.Count; i++)
+            {
+                TimeZoneInfo previous = zones[i - 1];
+                TimeZoneInfo current = zones[i];
+
+                int baseOffsetsCompared = current.BaseUtcOffset.CompareTo(previous.BaseUtcOffset);
+                Assert.True(baseOffsetsCompared >= 0,
+                    string.Format($"TimeZoneInfos are out of order. {previous.Id}:{previous.BaseUtcOffset} should be before {current.Id}:{current.BaseUtcOffset}"));
+
+                if (baseOffsetsCompared == 0)
+                {
+                    Assert.True(string.CompareOrdinal(current.DisplayName, previous.DisplayName) >= 0,
+                        string.Format($"TimeZoneInfos are out of order. {previous.DisplayName} should be before {current.DisplayName}"));
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
```

Comment:
Do we need `RemoteExecutor`? I'm not very familiar with it, but if I remember correctly this test will be skipped on Android and maybe other mobile platforms.

Reply:
Yes, it is important to do it with remote executions for two reasons. First, I want to exercise the test with ensuring GetSystemTimeZones is not called before in the process. Second, I am changing the process state by clearing the cached data which may affect any other existing or future tests that test the caches. I have added another test that execute without the remote executor to ensure some test coverage on Android or similar platforms.